### PR TITLE
Feature/error mod fixes

### PR DESCRIPTION
**Pull request by bbakernoaa**
@lholland-noaa

### CODE REVIEW

**Comment by bbakernoaa** [comment]
@zmoon please take a look the Lint steps keep failing

File: src/core/CMakeLists.txt [hunk 1]
```diff
@@ -1,10 +1,13 @@
 set(_srcs
-    charpak_mod.F90
+    precision_mod.F90
```
Comment (zmoon) [inline src/core/CMakeLists.txt]: `precision_mod.F90` it was already in the list. better to keep alphabetical order to more easily avoid this duplication

**Comment by zmoon** [comment]
@bbakernoaa I think you should be safe to run `pre-commit run --all-files` and commit the result. Mainly due to files added in previous PRs, which were also failing.

File: src/core/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
     constants.F90
+    charpak_mod.F90
     diagnostic_mod.F90
     error_mod.F90
     precision_mod.F90
     species_mod.F90
-    qfyaml_mod.F90)
+    qfyaml_mod.F90
+    init_mod.F90
+    test_main.F90)
```
Comment (zmoon) [inline src/core/CMakeLists.txt:10]: This program (`test_main`) isn't part of the library. We can move to a separate `tests` directory.

Comment (bbakernoaa) [reply src/core/CMakeLists.txt:10]: @zmoon agree just building things out right now.  Constantly cleaning up and adding

Comment (zmoon) [reply src/core/CMakeLists.txt:10]: sure sure but the sooner we set it up, easier to add more

Comment (bbakernoaa) [reply src/core/CMakeLists.txt:10]: started moving it

**Pull request merged**